utils/build_ds_container.py

```diff
@@ -0,0 +1,257 @@
+                log.info('Reusing existing namespace: %s', namespace)
+            else:
+                raise err
+
+
+def supports_custom_resource(resource_name):
+    with client.ApiClient() as api_client:
+        api = client.ApiextensionsV1Api(api_client)
+        try:
+            api.read_custom_resource_definition(resource_name)
+            return True
+        except exceptions.ApiException as err:
+            return False
+            if err.status == 404:
+                return False
+            log.warning(
+                'Unable to determine if %s custom resources are supported in '
+                'this deployment', resource_name)
+            return False
+
+
+def create_image_stream():
+    # FIXME(lbragstad) Fill this out using an openshift client, since we're
+    # dealing with an openshift-specific resource.
+    with client.ApiClient() as api_client:
+        body = client.V1CustomResourceDefinition(
+            api_version='image.openshift.io/v1',
+            kind='ImageStream',
+            metadata={'name': 'openscap-ocp4-ds'},
+            spec={'lookupPolicy': {'local': True}})
+
+        api = client.ApiextensionsV1Api(api_client)
+        api.create_custom_resource_definition(body)
+
+
+def create_build_config(build_config):
+    # FIXME(lbragstad) Fill this out using an openshift client, since we're
+    # dealing with an openshift-specific resource.
+    pass
+
+
+def build_content_locally(products, debug=False):
+    build_binary_path = os.path.join(REPO_PATH, 'build_product')
+    command = [build_binary_path] + products
+    capture_output = True
+    if debug:
+        command.append('--debug')
+        capture_output = False
+    subprocess.run(command, check=True, capture_output=capture_output)
+    log.info(f"Successfully built content for {', '.join(args.products)}")
+
+
+def build_content_remotely(products):
+    pass
+
+
+# FIXME(lbragstad): This should use something like podman-py.
+def build_container(debug=False):
+    dockerfile = (
+        REPO_PATH + '/Dockerfiles/compliance_operator_content.Dockerfile')
+    command = [
+        'podman', 'build', '-f', dockerfile, '-t', 'localcontentbuild:latest',
+        '.']
+    capture_output = True
+    if debug:
+        capture_output = False
+    subprocess.run(command, check=True, capture_output=capture_output)
+    log.info('Successfully built container image')
+
+
+# FIXME(lbragstad): This should use something like podman-py.
+def push_container(repository, tag, debug=False):
+    repository_string = repository + ':' + tag
+    command = [
+        'podman', 'push', 'localhost/localcontentbuild:latest',
+        repository_string]
+    capture_output = True
+    if debug:
+        capture_output = False
+    result = subprocess.run(command, capture_output=capture_output)
+    if result.returncode == 0:
+        log.info(f'Pushed image to {repository}:{tag}')
+    elif result.returncode == 125:
+        log.error(
```

Comment:
Responded in the slack thread with some thoughts on the repo push.. It's unclear to me how the old script inherits repo and image tags to push this to the cluster.

Reply:
@JAORMX added that functionality recently  as command line arguments. In that implementation, the script relied on `podman login`.

I initially implemented this method with the ability to pass in a username/password and invoking the command with `--creds`, but afterwards I removed it to keep things simpler.

Instead, @JAORMX and I agreed that we can leave authentication to the user, prior to them running this script. In that case, I think the best we can do is just detect if authentication failed for some reason and give them guidance on how to fix it.